test/Groups-conformance-tests.jl

```diff
@@ -0,0 +1,298 @@
+function test_Group_interface(G::Group)
+    @testset "Group interface" begin
+#        @testset "Iteration protocol" begin
+#            IS = Base.IteratorSize(typeof(G))
+#            if IS isa Base.IsInfinite
+#                @test isfinite(G) == false
+#            else
+#                isfiniteG = false
+#                if IS isa Base.HasLength || IS isa Base.HasShape
+#                    @test isfinite(G) == true
+#                    isfiniteG = true
+#                else
+#                    @test IS isa Base.SizeUnknown
+#                    try
+#                        @test isfinite(G) isa Bool
+#                        isfiniteG = isfinite(G)
+#                    catch e
+#                        @test e isa AbstractAlgebra.InfiniteOrder
+#                        isfiniteG = false
+#                    end
+#                end
+#
+#                if isfiniteG
+#                    @test length(G) isa Int
+#                    @test length(G) > 0
+#
+#                    @test eltype(G) <: GroupElem
+#                    @test one(G) isa eltype(G)
+#
+#                    if hasgens(G)
+#                        @test first(iterate(G)) isa eltype(G)
+#                        _, s = iterate(G)
+#                        @test first(iterate(G, s)) isa eltype(G)
+#                        @test isone(first(G))
+#                    end
+#                end
+#            end
+#        end
+
+        @testset "Group generators" begin
+            @test hasgens(G) isa Bool
+
+            if hasgens(G)
+                @test ngens(G) isa Int
+                @test gens(G) isa AbstractVector{eltype(G)}
+                @test length(gens(G)) == ngens(G)
+                if ngens(G) > 0
+                    @test first(gens(G)) == gen(G, 1)
+                    @test last(gens(G)) == gen(G, ngens(G))
+                end
+            else
+                # TODO: throw something more specific
+                @test_throws ErrorException gens(G)
+                @test_throws ErrorException ngens(G)
+            end
+        end
+
+        @testset "order, rand" begin
+            if isfinite(G)
+                @test order(Int16, G) isa Int16
```

Comment:
This test case seems problematic; what if `G` has order greater than `typemax(Int16)`? Probably should instead first get `order(BigInt, G)` and check if that is small enough to fit into an `Int16` (and if it doesn't fit, then perhaps it should `@test_throws order(Int16, G)` instead?)